server/shadow/Mac/mac_shadow.c

```diff
@@ -381,14 +382,16 @@ static void (^mac_capture_stream_handler)(
   if (count < 1)
 	  return;
 
+  EnterCriticalSection(&(surface->lock));
   mac_shadow_capture_get_dirty_region(subsystem);
   surfaceRect.left = 0;
   surfaceRect.top = 0;
   surfaceRect.right = surface->width;
   surfaceRect.bottom = surface->height;
   region16_intersect_rect(&(surface->invalidRegion), &(surface->invalidRegion), &surfaceRect);
+  empty = region16_is_empty(&(surface->invalidRegion); LeaveCriticalSection(&(surface->lock));
```

Comment:
Formatting (the following if looks off too ... wondering how this got through the checker)

Reply:
oh, that is not the formatting...